### ISSUE TITLE
Authenticated User Policy

### ISSUE THREAD

**Issue by youcefjd** [supabase-community/supabase-py#373]
Hey,

I set up a policy on a table to allow only authenticated users to insert into that specific table. 

I authenticated using the following code:

`supabase.auth.sign_in_with_password({"email": 'email@gmail.com', "password": '1234567'})`

and confirmed that there was an active session by running:

`supabase.auth.get_session()`

but when I ran an insert operation:

```
error, result = asyncio.run (
      supabase.table("user")
      .insert([{"prompt": 'hello', "id" : 0}])
)
```
I get the following error:

`SupabaseError: {'code': '42501', 'details': None, 'hint': None, 'message': 'new row violates row-level security policy for table "garbage"'}`

Should I specify the user's credentials in the insert statement so it can understand that there is an active session going on? How does it work exactly? 

Thank you!

**Comment by J0** [comment]
Hey @youcefjd,

The access token needs to be set - Postgrest then checks the access token to determine if user should be allowed to access the rows. I'm [working on a fix](https://github.com/supabase-community/supabase-py/pull/370) which will probably go out this weekend

**Comment by kamrik** [comment]
Seems to be the same issue as #185 
I found the following workaround to work well for me
```python
sb.auth.sign_in_with_password({"email": 'email@gmail.com', "password": '1234567'})
session = sb.auth.get_session()  # this refreshes the access token if needed
sb.postgrest.auth(token=session.access_token) # this sets the right "authorization" header on HTTP requests to PostgREST
data = sb.table('notes').select("body").execute()
```

**Comment by Lebusco-Fleet-Manager** [comment]
> Seems to be the same issue as #185 I found the following workaround to work well for me
> 
> ```python
> sb.auth.sign_in_with_password({"email": 'email@gmail.com', "password": '1234567'})
> session = sb.auth.get_session()  # this refreshes the access token if needed
> sb.postgrest.auth(token=session.access_token) # this sets the right "authorization" header on HTTP requests to PostgREST
> data = sb.table('notes').select("body").execute()
> ```

Its working for me.

**Comment by alvaroben** [comment]
> Seems to be the same issue as #185 I found the following workaround to work well for me
> 
> ```python
> sb.auth.sign_in_with_password({"email": 'email@gmail.com', "password": '1234567'})
> session = sb.auth.get_session()  # this refreshes the access token if needed
> sb.postgrest.auth(token=session.access_token) # this sets the right "authorization" header on HTTP requests to PostgREST
> data = sb.table('notes').select("body").execute()
> ```

Its working for me!

**Comment by anand2312** [comment]
Fixed by #560

**Comment by shawnesquivel** [comment]
Would be great if this could be added to the README? I'm still getting errors.
